### DISCUSSION TITLE
Consider gif/video on top of README

### DISCUSSION THREAD

**Discussion by tuurep** [jannis-baum/Vivify#164] [Ideas]
One thing that is so great about https://github.com/iamcco/markdown-preview.nvim is the gif on the readme, where you can understand everything by just seeing that.

I showed Vivify to one of my friends and he thought we should have a gif too :smile:

@jannis-baum @tweekism what do you think about this idea?

**Comment by tuurep** [comment]
Oh and I specifically mean demoing Vim/Neovim usage although viv can be used more generally, this is the most *demoable* scenario.

**Comment by Tweekism** [comment]
I am a big fan of the readme gifs

**Comment by jannis-baum** [comment]
Same here, big fan. Would be great to have one. The idea of showcasing the live preview makes sense. Maybe with some of the "additional" features.

Should we add the same gif to the vim plugin repo as well?

**Reply by tuurep** [reply]
Yeah I'm ideating in my head that it could be like:
* Show scrolling and editing on this repo's README.md
* Click on the directory-up button, navigate to CONTRIBUTING.md
* Do sumthin

> Should we add the same gif to the vim plugin repo as well?

Sure!

**Reply by jannis-baum** [reply]
Sounds good! Only thing I would change is going to the `tests/rendering/markdown-additional.md` instead of the `CONTRIBUTING.md` so that the GIF also has some fancy looking stuff in it ^^

**Reply by tuurep** [reply]
Ok, good idea

I could otherwise record one but I'm quite heavy on browser and nvim customization so some irrelevant stuff might draw attention :smile:

Would one of you guys be interested in recording?

I could still look into using a clean nvim rc + clean firefox profile, if you wish.

**Reply by jannis-baum** [reply]
I have a pretty minimal setup I think, if I record it would look something like this:

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/17814763-14d4-4340-a021-b3c48d893d32">

**Reply by tuurep** [reply]
Looks very nice for a gif IMO

Why no new back icon? :smile:

Here's my setup

![image](https://github.com/user-attachments/assets/e934d02c-c784-46f3-b6dc-1eb4efa446d8)

Actually, maybe it doesn't look that strange. Combining dark and light theme could be a cardinal sin for some.

**Reply by jannis-baum** [reply]
Haha, you caught me there on the icon🙈 I think I was doing something on some other branch before and hadn't rebased yet

Oh, interesting! I thought you'd have light mode in Vim as well ^^ Actually, I think we could do 2 versions of the GIF, one for dark and one for light mode and have the correct one show up on GitHub based on the GitHub theme. Do you have a light color theme for Vim?

**Reply by tuurep** [reply]
:smile: 

> have the correct one show up on GitHub based on the GitHub theme.

Oh, I didn't know it's possible to have a README segment that's conditional on something like that (are you sure it is?)

> Do you have a light color theme for Vim?

No and I've never used any, but could always install some light theme for the purpose of the gif

**Reply by Tweekism** [reply]
@tuurep Here is an example of light / dark theme auto select thingy on GitHub (second screenshot)

https://github.com/Tweekism/iso-clock

**Reply by jannis-baum** [reply]
> No and I've never used any, but could always install some light theme for the purpose of the gif

Okay then maybe I'll do that and record both so they look more similar. I'll do it in the next days

**Reply by tuurep** [reply]
Ok, thanks!

**Comment by jannis-baum** [comment]
I still haven't gotten to record this but one thing we can already think about in the meantime is where to store the GIF. The two options that come to my mind are

1. In the `docs/`. This is the simplest option but means that anyone who clones the main branch has to download the (potentially large) GIF.
2. On a separate orphan branch that is just for showcase assets. Explained [here](https://gist.github.com/joncardasis/e6494afd538a400722545163eb2e1fa5) and I did this for [another repo of mine that you can check as an example](https://github.com/jannis-baum/fzf-dotfiles/tree/main). Benefit is that it keeps `main` tidy & small, potentially quite significant disadvantage is that Vivify doesn't understand the syntax (i.e. couldn't display the GIF) and it wouldn't make any sense/be feasible to support it ^^

**Reply by tuurep** [reply]
Yeah! Btw, I was also wondering that about the images in the md rendering test file, they're probably needlessly large:

```
$ du -sh tests/rendering/images
1.8M	tests/rendering/images
```

**Reply by jannis-baum** [reply]
True! Maybe we should compress one by a lot and switch the other one out for an SVG. The SVG will inherently be very small and it's also something that's good to test anyways because SVGs are plain text files whose MIME type has to be recognized correctly

**Reply by jannis-baum** [reply]
@tuurep what do you think about the original question?

**Reply by tuurep** [reply]
Sorry I missed that I shoulda given an opinion on it :smile: 

Yes for sure method 2 is good especially because you already know how

If we're worried that we are going to demo using the README and it will show a gif that doesn't render, we can of course demo on some other file than the README :smile: Really doesn't matter which.

But the idea of showing `tests/rendering/markdown-additional.md` at some point was really good

**Reply by jannis-baum** [reply]
True, I wasn't even thinking about the demo per se. Just in general the fact that Vivify won't be able to display its own README if we go for method 2😁

**Reply by tuurep** [reply]
Ah, didn't realize it's able to view gifs otherwise? Not something I ever ran into yet :smile:

Edit: yeah indeed it does

Yes actually it's a good point.

Well, I'd be fine either way.

**Reply by tuurep** [reply]
Looking at the first gif I could find the size was 176K so my huge images are definitely worse here.. :D

**Reply by jannis-baum** [reply]
I still haven't gotten to do this but I realized there's an easy solution to use the `assets` branch and still have Vivify be able to display it's README: We just put the GitHub link into the Markdown file, then it will fetch it from there as long as there's internet
